action/src/main/java/com/adyen/checkout/action/ActionDelegateProvider.kt

```diff
@@ -44,22 +46,42 @@ internal class ActionDelegateProvider {
     ): ActionDelegate {
         val delegate = when (action) {
             is AwaitAction -> {
-                AwaitComponent.PROVIDER.getDelegate(getConfiguration(configuration), savedStateHandle, application)
+                AwaitComponentProvider(parentConfiguration).getDelegate(
+                    getConfiguration(configuration),
```

Comment:
This line is a bit confusing, I had to look up `getConfiguration` before I understood. Maybe we should rename `getConfiguration` to something like `getSpecificConfiguration` or `getActionConfiguration`?

Reply:
Good point 👍